docs/source/pytest.rst

```diff
@@ -0,0 +1,436 @@
+The :py:class:`~cocotb_tools.runner.Runner` is fully configurable by using ``--cocotb-*`` command line arguments,
+configuration files like `pyproject.toml`_ or `fixture`_ arguments.
+
+The plugin provides an :fixture:`~cocotb_tools.pytest.plugin.hdl` fixture that will create a new instance of
+:py:class:`~cocotb_tools.pytest.hdl.HDL` that can be customized and then used in tests.
+
+An example is provided below, located in a project ``conftest.py`` file:
+
+.. code:: python
+
+    import pytest
+    from cocotb_tools.pytest.hdl import HDL
+
+
+    @pytest.fixture(name="sample_module")
+    def sample_module_fixture(hdl: HDL) -> HDL:
+        """Define HDL design by adding HDL source files to it.
+
+        Args:
+            hdl: Fixture created by the cocotb pytest plugin, representing a HDL design.
+
+        Returns:
+            Representation of HDL design with added HDL source files.
+        """
+        hdl.sources = (
+            # List HDL source files,
+            "sample_module.sv",
+        )
+
+        # Build HDL design
+        hdl.build()
+
+        return hdl
+
+
+.. currentmodule:: cocotb_tools
+
+
+:py:deco:`pytest.mark.cocotb`
+==============================
+
+The plugin provides the marker :py:deco:`pytest.mark.cocotb` which allows
+to configure all aspects of cocotb test and cocotb runner.
+
+.. code:: python
+
+    @pytest.mark.cocotb(timescale=("1ns", "1ps"))
+    def test_dut_using_different_timescale(sample_module: HDL) -> None:
+        """Test DUT using different timescale."""
+        sample_module.test()
+
+    @pytest.mark.cocotb(timeout=(200, "ns"))
+    async def test_dut_feature_with_timeout(dut) -> None:
+        """Test DUT feature. It must finish within 200 nanoseconds."""
+
+Additionally, positional arguments of :py:deco:`pytest.mark.cocotb` marker are equivalent to
+``test_module`` argument from :py:meth:`.Runner.test`.
+
+.. code:: python
+
+    @pytest.mark.cocotb("test_dut_tb_1", "test_dut_tb_2")
+    def test_dut_using_different_testbenches(sample_module: HDL) -> None:
+        """Use cocotb tests from ``test_dut_tb_1.py`` and ``test_dut_tb_2.py`` files to test DUT."""
+        sample_module.test()
+
+If no positional arguments were provided to :py:deco:`pytest.mark.cocotb`,
+plugin will load current Python module where :py:deco:`pytest.mark.cocotb` was used as cocotb testbench (Python file with
+cocotb tests).
+
+.. code:: python
+
+    @pytest.mark.cocotb
+    def test_dut_using_default_testbench(sample_module: HDL) -> None:
+        """Test DUT with cocotb tests defined in the same Python file as this test function."""
+        sample_module.test()
+
+    async def test_dut_feature_1(dut) -> None:
+        """Test DUT feature 1."""
+
+If ``toplevel`` argument is empty/non-set, plugin will use name of first test module but without
+``test_*`` prefix or ``*_test`` suffix. For example, if test module was ``test_dut`` then
+name of HDL top level design will be ``dut``.
```

Comment:
Let's not re-use ``dut`` here, to help differentiate from the fixture of that name.
```suggestion
``test_*`` prefix or ``*_test`` suffix. For example, if test module was ``test_design`` then
name of HDL top level design will be ``design``.
```